### PR TITLE
Quell some clang warnings

### PR DESCRIPTION
**Pull request by bpfoley**

### CODE REVIEW

**Comment by rgov** [comment]
Hi @bpfoley, thanks for the pull request. Just to warn you, in the past @msoos has been extremely hesitant to take any code changes because the code is very fragile.

File: minisat/utils/System.cc [hunk 1]
```diff
@@ -87,11 +87,11 @@ double Minisat::memUsed() {
     malloc_statistics_t t;
     malloc_zone_statistics(NULL, &t);
     return (double)t.max_size_in_use / (1024*1024); }
-double Minisat::memUsedPeak(bool strictlyPeak) { return memUsed(); }
+double Minisat::memUsedPeak(bool /*strictlyPeak*/) { return memUsed(); }
```
Comment (rgov) [inline minisat/utils/System.cc:90]: I think the correct way to do this would be to add `__attribute__((unused))` .

**Comment by bpfoley** [comment]
> Hi @bpfoley, thanks for the pull request. Just to warn you, in the past @msoos has been extremely hesitant to take any code changes because the code is very fragile.

Thanks for the note. I'll keep it in mind. What sort of fragility did you see?

**Comment by bpfoley** [comment]
I'm happy to use whatever convention you like for annotating unused args. That said, commenting out the arg name is already used in the code a couple of times and `__attribute__((unused))` isn't used at all:

```
System.cc:134:void Minisat::limitMemory(uint64_t /*max_mem_mb*/)
System.cc:155:void Minisat::limitTime(uint32_t /*max_cpu_time*/)
```

**Comment by msoos** [comment]
As per @rgov -- it's super-fragile. However, I have to say you made very careful changes. I'll accept these ones. Please be aware that it's extremely weird code, most cleanups I have seen have introduced very subtle speed issues that are super-hard to spot. Indeed, I have found a speed optimization once that even one of the best in the field didn't spot for years (and I didn't either). I'll take these for now, but please be aware that more complicated changes (e.g. function re-writes, loop optimizations, minor changes like reversing a loop order, etc.) can have very weird and strong effects. So beware :)

**Pull request merged**